src/aks-preview/azext_aks_preview/tests/latest/custom_preparers.py

```diff
@@ -4,59 +4,36 @@
 # --------------------------------------------------------------------------------------------
 
 import os
-from datetime import datetime
 
-from azure.cli.testsdk import CliTestError
-from azure.cli.testsdk.preparers import NoTrafficRecordingPreparer
-from azure_devtools.scenario_tests import SingleValueReplacer
-from azure.cli.testsdk.reverse_dependency import get_dummy_cli
-
-
-class AKSCustomResourceGroupPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
-    def __init__(self, name_prefix='clitest.rg',
-                 parameter_name='resource_group',
-                 parameter_name_for_location='resource_group_location', location='westus',
-                 dev_setting_name='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_NAME',
-                 dev_setting_location='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_LOCATION',
-                 random_name_length=75, key='rg'):
-        if ' ' in name_prefix:
-            raise CliTestError(
-                'Error: Space character in resource group name prefix \'%s\'' % name_prefix)
+from azure.cli.testsdk.preparers import ResourceGroupPreparer
+
+
+class AKSCustomResourceGroupPreparer(ResourceGroupPreparer):
+    def __init__(
+        self,
+        name_prefix="clitest.rg",
+        parameter_name="resource_group",
+        parameter_name_for_location="resource_group_location",
+        location="westus",
+        dev_setting_name="AZURE_CLI_TEST_DEV_RESOURCE_GROUP_NAME",
+        dev_setting_location="AZURE_CLI_TEST_DEV_RESOURCE_GROUP_LOCATION",
+        random_name_length=75,
+        key="rg",
+    ):
         super(AKSCustomResourceGroupPreparer, self).__init__(
-            name_prefix, random_name_length)
-        self.cli_ctx = get_dummy_cli()
-        self.location = location
-        self.parameter_name = parameter_name
-        self.parameter_name_for_location = parameter_name_for_location
-        self.key = key
+            name_prefix,
+            parameter_name,
+            parameter_name_for_location,
+            location,
+            dev_setting_name,
+            dev_setting_location,
+            random_name_length,
+            key,
+        )
 
```

Comment:
You already add custom preparer in azure cli repo. consider importing form azure.cli repo.

Reply:
Sure, the custom preparer in extensions will be removed in the following [PR#3389](https://github.com/Azure/azure-cli-extensions/pull/3389).

Reply:
Whoops, it seems that this [change](https://github.com/Azure/azure-cli-extensions/pull/3389/commits/39360ba8f8c1cdc14d79d31f42a24715a4c56dfd) cannot pass the [integration test](https://dev.azure.com/azure-sdk/public/_build/results?buildId=908871&view=logs&j=9c80867e-ccda-50f1-65bc-e11447f3cda9&t=8e28b004-aba6-5a56-67c2-9d40a66e3ed9). In integration test, azure-cli is installed through pip and the test directory under acs (where the custom preparer resides) is not included.

Reply:
Maybe I should open another PR in cli to move the custom preparer out of the test directory？

Reply:
@Juliehzl